### DISCUSSION TITLE
Quarkus MongoDB Unit Tests - Init Scripts

### DISCUSSION THREAD

**Discussion by nawaz081** [quarkusio/quarkus#39774] [Q&A]
Hello,

I want to know if there is a way I can run init scripts before the Unit Tests for Mongo DB in Quarkus. I am running the tests using the dev services and I didn't find anything related to that in the dev services properties.

Thanks!

[a comment by an automated account is not shown]

**Reply by loicmathieu** [reply]
Hi,
You can do this with the Liquibase MongoDB extension: https://quarkus.io/guides/liquibase-mongodb

There is no facility to run an arbitrary MongoDB script (last time I checked this was removed from the driver so not easy to implement).

**Reply by nawaz081** [reply]
Thanks for the response. I will try and see if this works out.

**Reply by nawaz081** [reply]
@loicmathieu I have included the liquidbase config in my application.properties test folder. So the init scripts are executed and data is inserted but to a different mongodb cluster than the one created through quarkus dev services. I will dig further into it to see if there is any property in liquid base to use the default connection.

**Reply by loicmathieu** [reply]
It should use the default connection, if not it seems like a bug
